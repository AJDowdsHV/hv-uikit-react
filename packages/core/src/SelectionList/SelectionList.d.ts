import * as React from "react";
import { StandardProps } from "@material-ui/core";
import { HvFormElementProps } from "..";

export type HvSelectionListClassKey =
  | "root"
  | "label"
  | "description"
  | "listbox"
  | "vertical"
  | "horizontal"
  | "error";

export interface HvSelectionListProps
  extends StandardProps<HvFormElementProps, HvSelectionListClassKey, "onChange"> {
  /**
   * Indicates whether the list items group's orientation is horizontal or vertical.
   *
   * Defaults to vertical.
   */
  orientation: "vertical" | "horizontal";

  /**
   * The list items that are part of the group.
   *
   * Their state will always be controlled by the group.
   * However the individual list item onChange callback will still be called if defined.
   */
  children: React.ReactNode;
}

export default function HvSelectionList(props: HvSelectionListProps): JSX.Element | null;
