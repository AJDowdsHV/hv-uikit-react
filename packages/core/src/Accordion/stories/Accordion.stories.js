import React, { useState, useMemo } from "react";
import { makeStyles } from "@mui/styles";
import {
  HvAccordion,
  HvListContainer,
  HvListItem,
  HvInput,
  HvButton,
  HvTable,
  HvTableHead,
  HvTableHeader,
  HvTableRow,
  HvTableBody,
  HvTableCell,
  useHvData,
  useHvTableSticky,
} from "../..";

export default {
  title: "Display/Accordion",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvAccordion } from '@hitachivantara/uikit-react-core'",
  },

  component: HvAccordion,
};

export const Main = () => {
  const useStyles = makeStyles(() => ({
    accordionSpacing: {
      "& > li": {
        paddingLeft: 32,
      },
    },
    accordionContainer: {
      maxWidth: 300,
    },
  }));
  const classes = useStyles();
  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="item1" label="Analytics" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Views</HvListItem>
          <HvListItem>Parameters</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="item2" label="System" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Settings</HvListItem>
          <HvListItem>Network</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="item3" label="Data" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Storage</HvListItem>
          <HvListItem>Memory</HvListItem>
        </HvListContainer>
      </HvAccordion>
    </div>
  );
};

export const Disabled = () => {
  const useStyles = makeStyles(() => ({
    accordionSpacing: {
      "& > li": {
        paddingLeft: 32,
      },
    },
    accordionContainer: {
      maxWidth: 300,
    },
  }));
  const classes = useStyles();
  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="disabled-item1" label="Analytics" headingLevel={3} disabled>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Views</HvListItem>
          <HvListItem>Parameters</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="disabled-item2" label="System" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Settings</HvListItem>
          <HvListItem>Network</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="disabled-item3" label="Data" headingLevel={3} disabled>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Storage</HvListItem>
          <HvListItem>Memory</HvListItem>
        </HvListContainer>
      </HvAccordion>
    </div>
  );
};

Disabled.parameters = {
  docs: {
    description: { story: "A common accordion with disable items. " },
  },
};

export const Controlled = () => {
  const useStyles = makeStyles((theme) => ({
    accordionSpacing: {
      "& > li": {
        paddingLeft: 32,
      },
    },
    accordionContainer: {
      maxWidth: 300,
    },
    buttonContainer: {
      "& button": {
        marginRight: theme.hvSpacing("xs"),
        marginBottom: theme.hvSpacing("md"),
      },
    },
  }));
  const classes = useStyles();
  const [expandedState, setExpandedState] = useState({ analytics: true, system: true, data: true });
  const handleToggle = (key) => {
    const newValue = { ...expandedState };
    newValue[key] = !newValue[key];
    setExpandedState(newValue);
  };
  const handleAll = (option) => {
    setExpandedState({ analytics: option, system: option, data: option });
  };
  return (
    <>
      <div className={classes.buttonContainer}>
        <HvButton category="secondary" onClick={() => handleToggle("analytics")}>
          Toggle Analytics
        </HvButton>
        <HvButton category="secondary" onClick={() => handleToggle("system")}>
          Toggle System
        </HvButton>
        <HvButton category="secondary" onClick={() => handleToggle("data")}>
          Toggle Data
        </HvButton>
        <HvButton category="secondary" onClick={() => handleAll(false)}>
          Close all
        </HvButton>
        <HvButton category="secondary" onClick={() => handleAll(true)}>
          Expand all
        </HvButton>
      </div>
      <div className={classes.accordionContainer}>
        <HvAccordion
          id="controlled-item1"
          label="Analytics"
          onChange={() => handleToggle("analytics")}
          expanded={expandedState.analytics}
        >
          <HvListContainer className={classes.accordionSpacing} interactive condensed>
            <HvListItem>Views</HvListItem>
            <HvListItem>Parameters</HvListItem>
          </HvListContainer>
        </HvAccordion>
        <HvAccordion
          id="controlled-item2"
          label="System"
          onChange={() => handleToggle("system")}
          expanded={expandedState.system}
        >
          <HvListContainer className={classes.accordionSpacing} interactive condensed>
            <HvListItem>Settings</HvListItem>
            <HvListItem>Network</HvListItem>
          </HvListContainer>
        </HvAccordion>
        <HvAccordion
          id="controlled-item3"
          label="Data"
          onChange={() => handleToggle("data")}
          expanded={expandedState.data}
        >
          <HvListContainer className={classes.accordionSpacing} interactive condensed>
            <HvListItem>Storage</HvListItem>
            <HvListItem>Memory</HvListItem>
          </HvListContainer>
        </HvAccordion>
      </div>
    </>
  );
};

Controlled.parameters = {
  docs: {
    description: { story: "An accordion controlled externally forcing expansion." },
  },
};

export const Form = () => {
  const useStyles = makeStyles(() => ({
    accordionContainer: {
      maxWidth: 300,
    },
    formContainer: {
      padding: "0 32px",
      "& > div": {
        paddingTop: 17,
      },
    },
  }));
  const classes = useStyles();
  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="personal" label="Personal Information" headingLevel={3}>
        <div className={classes.formContainer}>
          <HvInput id="input-name" label="Name" placeholder="Insert first name" required />
          <HvInput id="input-email" label="Email" placeholder="Insert your email" required />
          <HvInput id="input-phone" label="Phone" placeholder="Insert your phone number" />
          <HvInput id="input-extension" label="Extension" placeholder="Insert phone extension" />
          <HvInput id="input-country" label="Country" placeholder="Insert country name" />
          <HvInput id="input-province" label="City/Province" placeholder="Insert province name" />
        </div>
      </HvAccordion>
      <HvAccordion id="billing" label="Billing Address" headingLevel={3}>
        <div className={classes.formContainer}>
          <HvInput id="input-address" label="Address 1" placeholder="Insert first name" />
          <HvInput id="input-address2" label="Address 2" placeholder="Insert address" />
          <HvInput id="input-city" label="City" placeholder="Insert city name" />
          <HvInput id="input-state" label="State" placeholder="Insert state" />
          <HvInput id="input-code" label="Zip Code" placeholder="Insert code" />
        </div>
      </HvAccordion>
      <HvAccordion id="shipping" label="Shipping Address" headingLevel={3}>
        <div className={classes.formContainer}>
          <HvInput id="input-address-bill" label="Address 1" placeholder="Insert first name" />
          <HvInput id="input-address2-bill" label="Address 2" placeholder="Insert address" />
          <HvInput id="input-city-bill" label="City" placeholder="Insert city name" />
          <HvInput id="input-state-bill" label="State" placeholder="Insert state" />
          <HvInput id="input-code-bill" label="Zip Code" placeholder="Insert code" />
        </div>
      </HvAccordion>
    </div>
  );
};

Form.parameters = {
  docs: {
    description: { story: "An accordion composition with forms." },
  },
};

export const Typography = () => {
  const useStyles = makeStyles(() => ({
    accordionContainer: {
      width: "100%",
    },
  }));
  const classes = useStyles();

  const getData = () => [
    {
      id: 1,
      title: "A new hope",
      director: "George Lucas",
      producer: "Gary Kurzt",
      release: "1977-05-25",
    },
    {
      id: 2,
      title: "Empires strike back",
      director: "Irvin Kershner",
      producer: "Gary Kurzt",
      release: "1980-05-17",
    },
    {
      id: 3,
      title: "Return of the jedi",
      director: "Richard Marquand",
      producer: "Howard Kazanjian",
      release: "1983-05-25",
    },
  ];

  const getColumns = () => [
    {
      Header: "Title",
      accessor: "title",
      sticky: "left",
      width: 150,
      style: { textTransform: "capitalize" },
    },
    {
      Header: "Director",
      accessor: "director",
    },
    {
      Header: "Producer",
      accessor: "producer",
    },
    {
      Header: "Release Date",
      accessor: "release",
    },
  ];

  const data = useMemo(() => getData(), []);
  const columns = useMemo(() => getColumns(), []);

  const { getTableProps, getTableHeadProps, getTableBodyProps, prepareRow, headerGroups, rows } =
    useHvData(
      {
        columns,
        data,
        stickyHeader: true,
      },
      useHvTableSticky
    );

  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="film" label="Films" headingLevel={2}>
        <HvTable id="accordion-table" {...getTableProps()}>
          <HvTableHead {...getTableHeadProps()}>
            {headerGroups.map((headerGroup) => (
              <HvTableRow {...headerGroup.getHeaderGroupProps()}>
                {headerGroup.headers.map((col) => (
                  <HvTableHeader {...col.getHeaderProps()}>{col.render("Header")}</HvTableHeader>
                ))}
              </HvTableRow>
            ))}
          </HvTableHead>
          <HvTableBody {...getTableBodyProps()}>
            {rows.map((row) => {
              prepareRow(row);

              return (
                <HvTableRow {...row.getRowProps()}>
                  {row.cells.map((cell) => (
                    <HvTableCell {...cell.getCellProps()}>{cell.render("Cell")}</HvTableCell>
                  ))}
                </HvTableRow>
              );
            })}
          </HvTableBody>
        </HvTable>
      </HvAccordion>
    </div>
  );
};

Typography.parameters = {
  docs: {
    description: { story: "An accordion with a different button typography." },
  },
};
