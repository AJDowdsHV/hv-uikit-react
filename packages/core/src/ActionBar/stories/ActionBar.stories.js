import React from "react";
import { makeStyles } from "@material-ui/core";
import { HvActionContainer, HvButton, HvDropDownMenu } from "../..";

export default {
  title: "Patterns/Action Bar/Action Container",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvActionContainer } from '@hv/uikit-react-core/dist'"
  },
  component: HvActionContainer
};

export const Main = () => {
  const useStyles = makeStyles(theme => ({
    root: {
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      width: "1000px"
    },
    flexBox: {
      display: "flex"
    },
    separator: {
      marginRight: theme.hv.spacing.sm
    }
  }));

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <HvActionContainer id="ActionContainer">
        <div className={classes.flexBox}>
          <div className={classes.separator}>
            <HvButton category="ghost" onClick={() => console.log("Save action")}>
              Save
            </HvButton>
          </div>
          <div>
            <HvButton category="ghost" onClick={() => console.log("Cancel action")}>
              Cancel
            </HvButton>
          </div>
        </div>
        <div>
          <HvButton category="ghost" onClick={() => console.log("Help action")}>
            Help
          </HvButton>
        </div>
      </HvActionContainer>
    </div>
  );
};

export const DualAction = () => {
  const useStyles = makeStyles(theme => ({
    root: {
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      width: "1000px"
    },
    flexBox: {
      display: "flex"
    },
    separator: {
      marginRight: theme.hv.spacing.sm
    }
  }));

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <HvActionContainer id="ActionContainer">
        <div className={classes.flexBox}>
          <div className={classes.separator}>
            <HvButton category="ghost" onClick={() => console.log("Save action")}>
              Save
            </HvButton>
          </div>
          <div>
            <HvButton category="ghost" onClick={() => console.log("Cancel action")}>
              Cancel
            </HvButton>
          </div>
        </div>
      </HvActionContainer>
    </div>
  );
};

DualAction.story = {
  parameters: {
    docs: {
      storyDescription: "Showcasing the action bar pattern with only two actions."
    }
  }
};

export const VariedActionBar = () => {
  const useStyles = makeStyles(theme => ({
    root: {
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      width: "1000px"
    },
    flexBox: {
      display: "flex"
    },
    separator: {
      marginRight: theme.hv.spacing.sm
    }
  }));

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <HvActionContainer id="ActionContainer">
        <div className={classes.flexBox}>
          <div className={classes.separator}>
            <HvButton category="ghost" onClick={() => console.log("Save action")}>
              Save
            </HvButton>
          </div>
          <div>
            <HvDropDownMenu
              onClick={(e, item) => console.log(item.label)}
              dataList={[{ label: "Delete" }, { label: "Update" }]}
            />
          </div>
        </div>
        <div>
          <HvButton category="ghost" onClick={() => console.log("Help action")}>
            Help
          </HvButton>
        </div>
      </HvActionContainer>
    </div>
  );
};

VariedActionBar.story = {
  parameters: {
    docs: {
      storyDescription: "Showcasing the action bar pattern with only two actions."
    }
  }
};
