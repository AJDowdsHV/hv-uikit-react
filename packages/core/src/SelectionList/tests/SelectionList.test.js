/* eslint-env jest */

import React from "react";
import { mount } from "enzyme";
import { Main } from "../stories/SelectionList.stories";

import { HvSelectionList, HvProvider } from "../..";

describe("SelectionList", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = mount(
      <HvProvider>
        <Main />
      </HvProvider>
    );
  });

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper.find(HvSelectionList)).toMatchSnapshot();
  });

  it("should render the HvSelectionList", () => {
    const component = wrapper.find(HvSelectionList);
    expect(component.length).toBe(1);
  });
});
