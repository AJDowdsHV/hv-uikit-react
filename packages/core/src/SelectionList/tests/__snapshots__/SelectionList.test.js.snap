// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionList should render correctly 1`] = `
<WithStyles(HvSelectionList)
  id="main"
  label="Choose your favorite items"
  name="favorite"
  onChange={[Function]}
>
  <HvSelectionList
    classes={
      Object {
        "description": "HvSelectionList-description",
        "error": "HvSelectionList-error",
        "horizontal": "HvSelectionList-horizontal",
        "label": "HvSelectionList-label",
        "listbox": "HvSelectionList-listbox",
        "root": "HvSelectionList-root",
        "selectAll": "HvSelectionList-selectAll",
        "vertical": "HvSelectionList-vertical",
      }
    }
    id="main"
    label="Choose your favorite items"
    name="favorite"
    onChange={[Function]}
  >
    <WithStyles(HvFormElement)
      className="HvSelectionList-root"
      disabled={false}
      id="main"
      name="favorite"
      readOnly={false}
      required={false}
      status="standBy"
    >
      <HvFormElement
        className="HvSelectionList-root"
        classes={
          Object {
            "root": "HvFormElement-root",
          }
        }
        disabled={false}
        id="main"
        name="favorite"
        readOnly={false}
        required={false}
        status="standBy"
      >
        <div
          className="HvSelectionList-root HvFormElement-root"
          id="main"
        >
          <WithStyles(HvLabel)
            className="HvSelectionList-label"
            id="main-label"
            label="Choose your favorite items"
          >
            <HvLabel
              className="HvSelectionList-label"
              classes={
                Object {
                  "childGutter": "HvLabel-childGutter",
                  "labelDisabled": "HvLabel-labelDisabled",
                  "root": "HvLabel-root",
                }
              }
              id="main-label"
              label="Choose your favorite items"
            >
              <WithStyles(ForwardRef)
                className="HvSelectionList-label HvLabel-root"
                component="label"
                id="main-label"
                variant="highlightText"
              >
                <ForwardRef
                  className="HvSelectionList-label HvLabel-root"
                  classes={
                    Object {
                      "3xlTitle": "HvTypography-3xlTitle",
                      "4xlTitle": "HvTypography-4xlTitle",
                      "5xlTitle": "HvTypography-5xlTitle",
                      "disabledText": "HvTypography-disabledText",
                      "highlightText": "HvTypography-highlightText",
                      "lTitle": "HvTypography-lTitle",
                      "link": "HvTypography-link",
                      "mTitle": "HvTypography-mTitle",
                      "noWrap": "HvTypography-noWrap",
                      "normalText": "HvTypography-normalText",
                      "placeholderText": "HvTypography-placeholderText",
                      "root": "HvTypography-root",
                      "sTitle": "HvTypography-sTitle",
                      "sectionTitle": "HvTypography-sectionTitle",
                      "selectedNavText": "HvTypography-selectedNavText",
                      "vizText": "HvTypography-vizText",
                      "vizTextDisabled": "HvTypography-vizTextDisabled",
                      "xlTitle": "HvTypography-xlTitle",
                      "xsInlineLink": "HvTypography-xsInlineLink",
                      "xsTitle": "HvTypography-xsTitle",
                      "xxlTitle": "HvTypography-xxlTitle",
                      "xxsTitle": "HvTypography-xxsTitle",
                    }
                  }
                  component="label"
                  id="main-label"
                  variant="highlightText"
                >
                  <label
                    className="HvSelectionList-label HvLabel-root HvTypography-root HvTypography-highlightText"
                    id="main-label"
                  >
                    Choose your favorite items
                  </label>
                </ForwardRef>
              </WithStyles(ForwardRef)>
            </HvLabel>
          </WithStyles(HvLabel)>
          <WithStyles(HvListContainer)
            aria-labelledby="main-label"
            className="HvSelectionList-listbox HvSelectionList-vertical"
            condensed={true}
            id="main-listbox"
            interactive={true}
            role="listbox"
          >
            <HvListContainer
              aria-labelledby="main-label"
              className="HvSelectionList-listbox HvSelectionList-vertical"
              classes={
                Object {
                  "root": "HvListContainer-root",
                }
              }
              condensed={true}
              id="main-listbox"
              interactive={true}
              role="listbox"
            >
              <ul
                aria-labelledby="main-label"
                className="HvSelectionList-listbox HvSelectionList-vertical HvListContainer-root"
                id="main-listbox"
                role="listbox"
              >
                <WithStyles(HvListItem)
                  key=".$.0"
                  onClick={[Function]}
                  role="option"
                  selected={false}
                  tabIndex={-1}
                  value="1"
                >
                  <HvListItem
                    classes={
                      Object {
                        "condensed": "HvListItem-condensed",
                        "disabled": "HvListItem-disabled",
                        "endAdornment": "HvListItem-endAdornment",
                        "focus": "HvListItem-focus",
                        "gutters": "HvListItem-gutters",
                        "interactive": "HvListItem-interactive",
                        "root": "HvListItem-root",
                        "selected": "HvListItem-selected",
                        "startAdornment": "HvListItem-startAdornment",
                        "withEndAdornment": "HvListItem-withEndAdornment",
                        "withStartAdornment": "HvListItem-withStartAdornment",
                      }
                    }
                    onClick={[Function]}
                    role="option"
                    selected={false}
                    tabIndex={-1}
                    value="1"
                  >
                    <WithStyles(Focus)
                      classes={
                        Object {
                          "focus": "HvListItem-focus",
                        }
                      }
                      configuration={
                        Object {
                          "tabIndex": -1,
                        }
                      }
                      rootRef={
                        Object {
                          "current": <ul
                            aria-labelledby="main-label"
                            class="HvSelectionList-listbox HvSelectionList-vertical HvListContainer-root"
                            id="main-listbox"
                            role="listbox"
                          >
                            <li
                              aria-selected="false"
                              class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvFocus-root HvFocus-focusDisabled"
                              role="option"
                              tabindex="-1"
                              value="1"
                            >
                              ListItem 1
                            </li>
                            <li
                              aria-selected="true"
                              class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvListItem-selected HvFocus-root HvFocus-selected HvFocus-focusDisabled"
                              role="option"
                              tabindex="0"
                              value="2"
                            >
                              ListItem 2
                            </li>
                            <li
                              aria-selected="false"
                              class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvFocus-root HvFocus-focusDisabled"
                              role="option"
                              tabindex="-1"
                              value="3"
                            >
                              ListItem 3
                            </li>
                          </ul>,
                        }
                      }
                      selected={false}
                      strategy="listbox"
                    >
                      <Focus
                        classes={
                          Object {
                            "disabled": "HvFocus-disabled",
                            "externalReference": "HvFocus-externalReference",
                            "falseFocus": "HvFocus-falseFocus",
                            "focus": "HvFocus-focus HvListItem-focus",
                            "focusDisabled": "HvFocus-focusDisabled",
                            "focused": "HvFocus-focused",
                            "root": "HvFocus-root",
                            "selected": "HvFocus-selected",
                          }
                        }
                        configuration={
                          Object {
                            "tabIndex": -1,
                          }
                        }
                        rootRef={
                          Object {
                            "current": <ul
                              aria-labelledby="main-label"
                              class="HvSelectionList-listbox HvSelectionList-vertical HvListContainer-root"
                              id="main-listbox"
                              role="listbox"
                            >
                              <li
                                aria-selected="false"
                                class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvFocus-root HvFocus-focusDisabled"
                                role="option"
                                tabindex="-1"
                                value="1"
                              >
                                ListItem 1
                              </li>
                              <li
                                aria-selected="true"
                                class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvListItem-selected HvFocus-root HvFocus-selected HvFocus-focusDisabled"
                                role="option"
                                tabindex="0"
                                value="2"
                              >
                                ListItem 2
                              </li>
                              <li
                                aria-selected="false"
                                class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvFocus-root HvFocus-focusDisabled"
                                role="option"
                                tabindex="-1"
                                value="3"
                              >
                                ListItem 3
                              </li>
                            </ul>,
                          }
                        }
                        selected={false}
                        strategy="listbox"
                      >
                        <ConditionalWrapper
                          condition={false}
                          wrapper={[Function]}
                        >
                          <li
                            aria-selected={false}
                            className="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvFocus-root HvFocus-focusDisabled"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            role="option"
                            selected={false}
                            value="1"
                          >
                            ListItem 1
                          </li>
                        </ConditionalWrapper>
                      </Focus>
                    </WithStyles(Focus)>
                  </HvListItem>
                </WithStyles(HvListItem)>
                <WithStyles(HvListItem)
                  key=".$.1"
                  onClick={[Function]}
                  role="option"
                  selected={true}
                  tabIndex={0}
                  value="2"
                >
                  <HvListItem
                    classes={
                      Object {
                        "condensed": "HvListItem-condensed",
                        "disabled": "HvListItem-disabled",
                        "endAdornment": "HvListItem-endAdornment",
                        "focus": "HvListItem-focus",
                        "gutters": "HvListItem-gutters",
                        "interactive": "HvListItem-interactive",
                        "root": "HvListItem-root",
                        "selected": "HvListItem-selected",
                        "startAdornment": "HvListItem-startAdornment",
                        "withEndAdornment": "HvListItem-withEndAdornment",
                        "withStartAdornment": "HvListItem-withStartAdornment",
                      }
                    }
                    onClick={[Function]}
                    role="option"
                    selected={true}
                    tabIndex={0}
                    value="2"
                  >
                    <WithStyles(Focus)
                      classes={
                        Object {
                          "focus": "HvListItem-focus",
                        }
                      }
                      configuration={
                        Object {
                          "tabIndex": 0,
                        }
                      }
                      rootRef={
                        Object {
                          "current": <ul
                            aria-labelledby="main-label"
                            class="HvSelectionList-listbox HvSelectionList-vertical HvListContainer-root"
                            id="main-listbox"
                            role="listbox"
                          >
                            <li
                              aria-selected="false"
                              class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvFocus-root HvFocus-focusDisabled"
                              role="option"
                              tabindex="-1"
                              value="1"
                            >
                              ListItem 1
                            </li>
                            <li
                              aria-selected="true"
                              class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvListItem-selected HvFocus-root HvFocus-selected HvFocus-focusDisabled"
                              role="option"
                              tabindex="0"
                              value="2"
                            >
                              ListItem 2
                            </li>
                            <li
                              aria-selected="false"
                              class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvFocus-root HvFocus-focusDisabled"
                              role="option"
                              tabindex="-1"
                              value="3"
                            >
                              ListItem 3
                            </li>
                          </ul>,
                        }
                      }
                      selected={true}
                      strategy="listbox"
                    >
                      <Focus
                        classes={
                          Object {
                            "disabled": "HvFocus-disabled",
                            "externalReference": "HvFocus-externalReference",
                            "falseFocus": "HvFocus-falseFocus",
                            "focus": "HvFocus-focus HvListItem-focus",
                            "focusDisabled": "HvFocus-focusDisabled",
                            "focused": "HvFocus-focused",
                            "root": "HvFocus-root",
                            "selected": "HvFocus-selected",
                          }
                        }
                        configuration={
                          Object {
                            "tabIndex": 0,
                          }
                        }
                        rootRef={
                          Object {
                            "current": <ul
                              aria-labelledby="main-label"
                              class="HvSelectionList-listbox HvSelectionList-vertical HvListContainer-root"
                              id="main-listbox"
                              role="listbox"
                            >
                              <li
                                aria-selected="false"
                                class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvFocus-root HvFocus-focusDisabled"
                                role="option"
                                tabindex="-1"
                                value="1"
                              >
                                ListItem 1
                              </li>
                              <li
                                aria-selected="true"
                                class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvListItem-selected HvFocus-root HvFocus-selected HvFocus-focusDisabled"
                                role="option"
                                tabindex="0"
                                value="2"
                              >
                                ListItem 2
                              </li>
                              <li
                                aria-selected="false"
                                class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvFocus-root HvFocus-focusDisabled"
                                role="option"
                                tabindex="-1"
                                value="3"
                              >
                                ListItem 3
                              </li>
                            </ul>,
                          }
                        }
                        selected={true}
                        strategy="listbox"
                      >
                        <ConditionalWrapper
                          condition={false}
                          wrapper={[Function]}
                        >
                          <li
                            aria-selected={true}
                            className="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvListItem-selected HvFocus-root HvFocus-selected HvFocus-focusDisabled"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            role="option"
                            selected={true}
                            value="2"
                          >
                            ListItem 2
                          </li>
                        </ConditionalWrapper>
                      </Focus>
                    </WithStyles(Focus)>
                  </HvListItem>
                </WithStyles(HvListItem)>
                <WithStyles(HvListItem)
                  key=".$.2"
                  onClick={[Function]}
                  role="option"
                  selected={false}
                  tabIndex={-1}
                  value="3"
                >
                  <HvListItem
                    classes={
                      Object {
                        "condensed": "HvListItem-condensed",
                        "disabled": "HvListItem-disabled",
                        "endAdornment": "HvListItem-endAdornment",
                        "focus": "HvListItem-focus",
                        "gutters": "HvListItem-gutters",
                        "interactive": "HvListItem-interactive",
                        "root": "HvListItem-root",
                        "selected": "HvListItem-selected",
                        "startAdornment": "HvListItem-startAdornment",
                        "withEndAdornment": "HvListItem-withEndAdornment",
                        "withStartAdornment": "HvListItem-withStartAdornment",
                      }
                    }
                    onClick={[Function]}
                    role="option"
                    selected={false}
                    tabIndex={-1}
                    value="3"
                  >
                    <WithStyles(Focus)
                      classes={
                        Object {
                          "focus": "HvListItem-focus",
                        }
                      }
                      configuration={
                        Object {
                          "tabIndex": -1,
                        }
                      }
                      rootRef={
                        Object {
                          "current": <ul
                            aria-labelledby="main-label"
                            class="HvSelectionList-listbox HvSelectionList-vertical HvListContainer-root"
                            id="main-listbox"
                            role="listbox"
                          >
                            <li
                              aria-selected="false"
                              class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvFocus-root HvFocus-focusDisabled"
                              role="option"
                              tabindex="-1"
                              value="1"
                            >
                              ListItem 1
                            </li>
                            <li
                              aria-selected="true"
                              class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvListItem-selected HvFocus-root HvFocus-selected HvFocus-focusDisabled"
                              role="option"
                              tabindex="0"
                              value="2"
                            >
                              ListItem 2
                            </li>
                            <li
                              aria-selected="false"
                              class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvFocus-root HvFocus-focusDisabled"
                              role="option"
                              tabindex="-1"
                              value="3"
                            >
                              ListItem 3
                            </li>
                          </ul>,
                        }
                      }
                      selected={false}
                      strategy="listbox"
                    >
                      <Focus
                        classes={
                          Object {
                            "disabled": "HvFocus-disabled",
                            "externalReference": "HvFocus-externalReference",
                            "falseFocus": "HvFocus-falseFocus",
                            "focus": "HvFocus-focus HvListItem-focus",
                            "focusDisabled": "HvFocus-focusDisabled",
                            "focused": "HvFocus-focused",
                            "root": "HvFocus-root",
                            "selected": "HvFocus-selected",
                          }
                        }
                        configuration={
                          Object {
                            "tabIndex": -1,
                          }
                        }
                        rootRef={
                          Object {
                            "current": <ul
                              aria-labelledby="main-label"
                              class="HvSelectionList-listbox HvSelectionList-vertical HvListContainer-root"
                              id="main-listbox"
                              role="listbox"
                            >
                              <li
                                aria-selected="false"
                                class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvFocus-root HvFocus-focusDisabled"
                                role="option"
                                tabindex="-1"
                                value="1"
                              >
                                ListItem 1
                              </li>
                              <li
                                aria-selected="true"
                                class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvListItem-selected HvFocus-root HvFocus-selected HvFocus-focusDisabled"
                                role="option"
                                tabindex="0"
                                value="2"
                              >
                                ListItem 2
                              </li>
                              <li
                                aria-selected="false"
                                class="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvFocus-root HvFocus-focusDisabled"
                                role="option"
                                tabindex="-1"
                                value="3"
                              >
                                ListItem 3
                              </li>
                            </ul>,
                          }
                        }
                        selected={false}
                        strategy="listbox"
                      >
                        <ConditionalWrapper
                          condition={false}
                          wrapper={[Function]}
                        >
                          <li
                            aria-selected={false}
                            className="HvListItem-root HvListItem-gutters HvListItem-condensed HvListItem-interactive HvFocus-root HvFocus-focusDisabled"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            role="option"
                            selected={false}
                            value="3"
                          >
                            ListItem 3
                          </li>
                        </ConditionalWrapper>
                      </Focus>
                    </WithStyles(Focus)>
                  </HvListItem>
                </WithStyles(HvListItem)>
              </ul>
            </HvListContainer>
          </WithStyles(HvListContainer)>
        </div>
      </HvFormElement>
    </WithStyles(HvFormElement)>
  </HvSelectionList>
</WithStyles(HvSelectionList)>
`;
